### DISCUSSION TITLE
Unfocused tab activity notification

### DISCUSSION THREAD

**Discussion by edevil** [ghostty-org/ghostty#2871] [Feature Requests, Ideas]
One feature that I miss from iTerm2 is the unfocused tab activity notification.

When we have multiple tabs open and there is some activity in one of the tabs that is not on the foreground, it would be great if we could be notified by some change in the tab title, for example.

I could not find anything related to this in the Ghostty documentation.

**Comment by franklouwers** [comment]
One of the first things I noticed as well. Love that feature in iTerm, would double-love to see it in Ghostty :)

**Comment by pjz** [comment]
I used to use this a lot in screen/tmux: start a longish build in one tab, enable silence monitoring (inverse of activity monitoring) and when the build stopped scrolling, I'd get a nice notification.  

Kitty has a config option `tab_activity_symbol` that will be displayed in a tab's titlebar when there's activity when it's not focused.  

I think both activity and silence monitoring are useful, though I'm unsure how best to implement the UI for them in a terminal context.

**Comment by TravisKangas** [comment]
i could drop tmux if this was implemented

**Comment by rhodes-b** [comment]
im fairly certain the bell feature handles this one of the default behaviors of the bell is to add a bell emoji into the titlebar

**Reply by 00-kat** [reply]
The bell feature adds an emoji to the titlebar *on a BEL character*, not on tab activity. The commenter above meant that `tab_activity_symbol` does it on tab activity; there's intersection between their functionality (they both add an emoji to the titlebar) but the emoji is added on different triggers.

**Comment by evmcl** [comment]
While the bell feature is useful, it is not the same. The use-case for an indicator upon output from an inactive or unfocused tab is running a long-running task that has sporadic output.

Without the indicator, you have to switch back to the tab periodically to see if there is anything new. With the indicator, you can just glance at the tab bar to see if you need to switch to check on how the task is going.
